lib/location/method_gnss.c

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdio.h>
+#include <zephyr.h>
+#include <logging/log.h>
+
+#include <modem/location.h>
+#include <nrf_modem_gnss.h>
+#include <nrf_errno.h>
+
+#include "loc_core.h"
+
+LOG_MODULE_DECLARE(location, CONFIG_LOCATION_LOG_LEVEL);
+
+extern location_event_handler_t event_handler;
+extern struct loc_event_data current_event_data;
+
+struct k_work gnss_fix_work;
+struct k_work method_gnss_timeout_work;
+
+void method_gnss_event_handler(int event)
+{
+	switch (event) {
+	case NRF_MODEM_GNSS_EVT_FIX:
+		LOG_DBG("GNSS: Got fix");
+		k_work_submit(&gnss_fix_work);
```

Comment:
There's a dedicated work queue for location API which we could utilize throughout the library:
`k_work_submit_to_queue(loc_core_work_queue_get(), &gnss_fix_work);`
@hiltunent

Reply:
This has been done